src/main/java/repositories/UserDataBase.java

```diff
@@ -0,0 +1,38 @@
+package repositories;
+
+import models.User;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class UserDataBase {
+    private static UserDataBase instance = new UserDataBase();
+    private Map<Integer, User> userMap = new HashMap<>();
+
+    private UserDataBase() {
+    }
+    public static synchronized UserDataBase getInstance(){
```

Comment:
Why do we need to make this method synchronized and check if the instance is null or not?
In line 12 you initialize instance.

Reply:
- [x] made DataBase class,  UserCounter classas and GameCounter as not Singleton
- [x] created GameService instance, UserCounter userCounter, GameCounter gameCounter, GameConstructor constructor, GameIterator iteratot, Map<Integer, AbstractGame> db  final fields

Reply:
- [x] fix lazy initialization in GameSerise.java

Reply:
- [x] fix lazy initialization for PasswordSaver.java and UserCounter.java too